Amazon.QLDB.Driver/client/Session.cs

```diff
@@ -87,29 +88,53 @@ internal static Session StartSession(string ledgerName, AmazonQLDBSessionClient
         }
 
         /// <summary>
-        /// Send an abort request to QLDB, rolling back any active changes and closing any open results.
+        /// Factory method for constructing a new Session, creating a new session to QLDB on construction.
+        /// </summary>
+        ///
+        /// <param name="ledgerName">The name of the ledger to create a session to.</param>
+        /// <param name="sessionClient">The low-level session used for communication with QLDB.</param>
+        /// <param name="logger">The logger to inject any logging framework.</param>
+        ///
+        /// <returns>A newly created <see cref="Session"/>.</returns>
+        internal static Session StartSession(string ledgerName, AmazonQLDBSessionClient sessionClient, ILogger logger)
+        {
+            return StartSessionAsync(ledgerName, sessionClient, logger).GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        /// Send an asynchronous abort request to QLDB, rolling back any active changes and closing any open results.
         /// </summary>
         ///
         /// <returns>The result of the abort transaction request.</returns>
-        internal virtual AbortTransactionResult AbortTransaction()
+        internal virtual async Task<AbortTransactionResult> AbortTransactionAsync()
         {
             var abortTransactionRequest = new AbortTransactionRequest();
             var request = new SendCommandRequest
             {
                 AbortTransaction = abortTransactionRequest,
             };
-            var response = this.SendCommand(request);
+            var response = await this.SendCommandAsync(request);
             return response.AbortTransaction;
         }
 
         /// <summary>
-        /// Send an end session request to QLDB and ignore exceptions.
+        /// Send an abort request to QLDB, rolling back any active changes and closing any open results.
         /// </summary>
-        internal virtual void End()
```

Comment:
The purpose of End is a convenient function that handles errors upon ending session (since we don't care about them).

We don't need EndAsync. It can just be `internal virtual async Task End()`. It awaits EndSession and catches the exception, logs and ignores it.

Reply:
Addressed.